magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleExtensions.kt

```diff
@@ -2,58 +2,58 @@ package com.wealthfront.magellan.lifecycle
 
 import kotlin.reflect.KProperty
 
-public fun <T : LifecycleAware, R> LifecycleOwner.lifecycle(lifecycleAware: T, getter: (T) -> R): Lifecycle<T, R> {
-  return Lifecycle(
+public fun <Child : LifecycleAware, Property> LifecycleOwner.attachFieldToLifecycle(lifecycleAware: Child, getter: (Child) -> Property): LifecycleChild<Child, Property> {
+  return LifecycleChild(
     this,
     lifecycleAware,
     getter
   )
 }
 
-public fun <T : LifecycleAware> LifecycleOwner.lifecycle(lifecycleAware: T): Lifecycle<T, T> {
-  return Lifecycle(
+public fun <Child : LifecycleAware> LifecycleOwner.attachFieldToLifecycle(lifecycleAware: Child): LifecycleChild<Child, Child> {
+  return LifecycleChild(
     this,
     lifecycleAware,
     { lifecycleAware }
   )
 }
 
-public class Lifecycle<T : LifecycleAware, R>(
+public class LifecycleChild<Child : LifecycleAware, Property>(
   parent: LifecycleOwner,
-  private var lifecycleAware: T,
-  public val getter: (T) -> R
+  private var lifecycleAware: Child,
+  public val valueGetter: (Child) -> Property
 ) {
 
-  private var overrideValue: R? = null
+  private var overrideValue: Property? = null
 
   init {
     parent.attachToLifecycle(lifecycleAware)
   }
 
-  public operator fun getValue(thisRef: Any?, property: KProperty<*>): R {
-    return overrideValue ?: getter(lifecycleAware)
+  public operator fun getValue(thisRef: Any?, property: KProperty<*>): Property {
+    return overrideValue ?: valueGetter(lifecycleAware)
   }
 
-  public operator fun setValue(thisRef: Any?, property: KProperty<*>, value: R) {
+  public operator fun setValue(thisRef: Any?, property: KProperty<*>, value: Property) {
     overrideValue = value
   }
 }
 
-public fun <CustomLifecycleAware : LifecycleAware> LifecycleOwner.lateinitLifecycle(): LateinitLifecycle<CustomLifecycleAware> {
-  return LateinitLifecycle(this)
+public fun <Child : LifecycleAware> LifecycleOwner.attachLateinitFieldToLifecycle(): LateinitLifecycleChild<Child> {
+  return LateinitLifecycleChild(this)
 }
```

Comment:
I like how `attachFieldToLifecycle` scales to `attachLateinitFieldToLifecycle` 👍